lang/py/src/avro/io.py

```diff
@@ -280,6 +297,79 @@ def read_utf8(self):
     """
     return unicode(self.read_bytes(), "utf-8")
 
+  def read_date_from_int(self):
+    """
+    Decode python date object from int
+    int stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    days_since_epoc = self.read_int()
+    return datetime.date(1970, 1, 1) + datetime.timedelta(days_since_epoc)
+
+  def read_time_millis_from_int(self):
+    """
+    Decode python time object from int
+    int stores the number of milliseconds after midnight, 00:00:00.000
+    """
+    milisec = self.read_int()
+    hrs = int(milisec / 3600000)
+    milisec = milisec % 3600000
+
+    minutes = int(milisec / 60000)
+    milisec = milisec % 60000
+
+    seconds = int(milisec / 1000)
+    milisec = milisec % 1000
+
+    microseconds = milisec * 1000
+    return datetime.time(
+      hour=hrs,
+      minute=minutes,
+      second=seconds,
+      microsecond=microseconds
+    )
```

Comment:
Have you looked into doing it simply by converting it to [timestamp](https://docs.python.org/2/library/datetime.html#datetime.date.fromtimestamp)? `datetime.datetime.fromtimestamp(ms/1000.0)`. I am not sure what are the performance implications of this but since its in the library I'd suggest looking into it.